src/server/game/Entities/Player/Player.cpp

```diff
@@ -4842,13 +4842,18 @@ void Player::RepopAtGraveyard()
     // Special handle for battleground maps
     if (Battleground* bg = GetBattleground())
         ClosestGrave = bg->GetClosestGraveyard(this);
-    else
+    else if (Battlefield* bf = sBattlefieldMgr->GetBattlefieldToZoneId(GetMap(), GetZoneId()))
+        ClosestGrave = bf->GetClosestGraveyard(this);
+    else if (InstanceScript* instance = GetInstanceScript())
     {
-        if (Battlefield* bf = sBattlefieldMgr->GetBattlefieldToZoneId(GetMap(), GetZoneId()))
-            ClosestGrave = bf->GetClosestGraveyard(this);
+        uint32 entrance = instance->GetEntranceLocation();
```

Comment:
just put that code into GetEntranceLocation when its 0, in that case you can just call GetEntranceLocation

Reply:
that might break scripts from people who are using this features in their own forks. That being said, maybe my previous attempt with ZoneScript wasn't so bad after all. Could have just overriden it for InstanceScript to call EntranceLocation if present else find graveyard 🤔

Reply:
Since when do we care about forks?